apps/rush-lib/src/logic/versionMismatch/VersionMismatchFinder.ts

```diff
@@ -96,20 +111,24 @@ export class VersionMismatchFinder {
     options: {
       isRushCheckCommand: boolean;
       variant?: string | undefined;
+      jsonFlag?: boolean | undefined;
     }
   ): void {
-
     if (rushConfiguration.ensureConsistentVersions || options.isRushCheckCommand) {
       const mismatchFinder: VersionMismatchFinder = VersionMismatchFinder.getMismatches(rushConfiguration, options);
 
-      mismatchFinder.print();
-
-      if (mismatchFinder.numberOfMismatches) {
-        console.log(colors.red(`Found ${mismatchFinder.numberOfMismatches} mis-matching dependencies!`));
-        process.exit(1);
+      if (options.jsonFlag) {
+        mismatchFinder.printAsJson();
       } else {
-        if (options.isRushCheckCommand) {
-          console.log(colors.green(`Found no mis-matching dependencies!`));
+        mismatchFinder.print();
+
+        if (mismatchFinder.numberOfMismatches > 0) {
+          console.log(colors.red(`Found ${mismatchFinder.numberOfMismatches} mis-matching dependencies!`));
+          process.exit(1);
```

Comment:
This should probably throw an `AlreadyReportedError` instead of exiting directly.